dip-0003.md

```diff
@@ -105,7 +105,7 @@ The transaction consists of the following data in the payload area:
 | scriptPayout | Script | Variable | Payee script (p2pkh/p2sh) |
 | inputsHash | uint256 | 32 | Hash of all the outpoints of the transaction inputs |
 | payloadSigSize | compactSize uint | 1-9 | Size of the Signature |
-| payloadSig | vector<unsigned char> | Variable | Signature of the hash of the ProTx fields. Signed with KeyIdOwner |
+| payloadSig | vector<unsigned char> | Variable | Signature of the hash of the ProTx fields. Signed with the key corresponding to the collateral outpoint. |
```

Comment:
I believe that we now need 2 sigs here. One for the collateral key, which is only set when the ProRegTx refers to another TX. And one that is always signed with the owner key. Otherwise there is no proof in the ProRegTx that the owner actually holds the owner key. As an alternative, we could force the KeyIdOwner to be equal to the key id from the collateral in case the collateral is from another TX.

Reply:
>Otherwise there is no proof in the ProRegTx that the owner actually holds the owner key. 

Hmmmm... But why would you need it? With ProRegTx you basically say `I have 1000 DASH (here or in another tx), here is the proof (the sig) and here are params I want to use to register my masternode`. And that should be it, no?

Reply:
Hmm true. Regarding `(here or in another tx)`, that only applies to referring another TX. In the case where the collateral is part of the ProRegTx, just the fact that the collateral was successfully moved is proof that he owns it. Also, in that case we would be able to allow multisig/P2SH if we left out verification of the sig here.

Reply:
```suggestion
| payloadSig | vector<unsigned char> | Variable | Signature of the hash of the ProTx fields. Signed with the key corresponding to the collateral outpoint in case the collateral is not part of the ProRegTx itself. |
```

Reply:
To clarify, does it mean that we leave the signature empty if the collateral is actually a part of the ProRegTx? Or do we sign it with keyIdOwner in this case? I guess there is no reason to do this but asking to make sure.

Reply:
Leaving it empty should be ok. We might need to add a description of this behavior somewhere else as well. Will put more time into this on monday.

Reply:
Maybe extend the suggestion above with `, empty otherwise` or smth like that?

Reply:
Updated the suggested change (btw, cool new Github feature)